libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1317,10 +1350,34 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context << Instruction::SUB;
 		break;
 	case Token::Mul:
-		m_context << Instruction::MUL;
+		if (c_isFractional)
+		{
+			//multiply, then shift right...this is your fraction.
```

Comment:
so after mulling it over, I think the only time when my solution does not work (and my solution is more efficient than what I would need to implement in your example (and I intend to implement that because...why the hell not...seems like it'd work for the problems I'm running into)) is when the fractional bits are larger than 128 and still contain an integer.